### PR TITLE
feat(CI): Initial workflow file

### PR DESCRIPTION
**Pull request by lmilbaum**

### CODE REVIEW

File: .github/workflows/blank.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# This is a basic workflow to help you get started with Actions
+
+name: CI
```
Comment (viceice) [inline .github/workflows/blank.yml:3]:
```suggestion
name: build
```

rename file to `build.yaml`

File: .github/workflows/blank.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the "main" branch
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
```
Comment (viceice) [inline .github/workflows/blank.yml:14]: add merge queue support

**Comment by viceice** [comment]
we should also create workflows like on containerbase, so we don't  run expensive docker builds twice on local PR (source branch is in this repo)

see here for sample
- https://github.com/containerbase/base/tree/main/.github/workflows

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - ready_for_review
+
+  merge_group:
+
+  workflow_dispatch:
```
Comment (viceice) [inline .github/workflows/build.yml:18]: doesn't work, you need three files.

File: .github/workflows/build-push.yml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+jobs:
+  build:
+    uses: ./.github/workflows/build.yml
+    permissions:
+      contents: read
+      checks: write
+    secrets:
+      DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+      DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
```
Comment (viceice) [inline .github/workflows/build-push.yml:34]: remove, we only push to GitHub

File: .github/workflows/build-push.yml [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+          git config --global core.autocrlf false
+          git config --global core.symlinks true
+          git config --global user.email 'renovate@whitesourcesoftware.com'
+          git config --global user.name  'Renovate Bot'
+          echo "CACHE_WEEK=$(date +%U)" >> $GITHUB_ENV
+
+      - name: Docker registry login
+        if: github.ref_name == 'main'
+        run: |
+          echo "${{ secrets.DOCKER_PASSWORD }}" | docker login -u ${{ secrets.DOCKER_USERNAME }} --password-stdin
```
Comment (viceice) [inline .github/workflows/build-push.yml:83]: remove

File: .github/workflows/build-push.yml [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+        run: yarn install
+
+      - name: prepare apt proxy
+        run: sudo yarn prepare:proxy
+
+      - name: semantic-release
+        if: github.event_name == 'push'
+        run: yarn semantic-release --dry-run ${{env.DRY_RUN}} --ci ${{env.DRY_RUN != 'true'}}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (viceice) [inline .github/workflows/build-push.yml:96]: comment out, otherwise workflow will fail

File: .github/workflows/build.yml [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+  push:
+    branches:
+      - main
+
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - ready_for_review
```
Comment (viceice) [inline .github/workflows/build.yml:14]: move to build-pr

File: .github/workflows/build.yml [hunk 8]
```diff
@@ -0,0 +1,33 @@
+name: build
+
+on:
+ 
+  push:
+    branches:
+      - main
```
Comment (viceice) [inline .github/workflows/build.yml:7]: remove

File: .github/workflows/build.yml (hunk 3 above)
Comment (viceice) [inline .github/workflows/build.yml:18]: move to build-push

**Comment by lmilbaum** [comment]
@viceice ping

**Review by viceice — APPROVED** [review]

**Pull request merged**